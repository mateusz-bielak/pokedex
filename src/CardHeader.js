import PropTypes from 'prop-types';
import React from 'react';
import styled from '@emotion/styled';

import { CardWrapper } from './CardWrapper';
import { breakpoints, colors, getImageSrc } from './variables';

const Header = styled.header`
    box-sizing: border-box;
    display: flex;
    align-items: flex-end;
    height: 20vw;
    width: 100%;
    margin-bottom: 10px;

    ${breakpoints.large} {
        height: 15vw;
        margin-bottom: 20px;
    }
`;

const NameCard = styled(CardWrapper)`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    height: 100%;
    width: 100%;
    margin-right: 5px;
    text-align: center;

    ${breakpoints.large} {
        margin-right: 20px;
    }
`;

const Name = styled.span`
    display: block;

    font-weight: bold;
    text-transform: capitalize;
`;

const Image = styled.img`
    box-sizing: border-box;
    max-height: 100%;
    padding: 5px;

    border-radius: 20px;
    background-color: ${colors.pokeballSecondary};

    filter: ${props => (props.blur ? 'blur(5px)' : 'none')};
`;

export const CardHeader = ({ blur, genus, name, onImageLoad, pokemonId }) => (
    <Header>
        <NameCard>
            <Name>{name}</Name>
            <span>{genus}</span>
        </NameCard>
        <Image src={getImageSrc(pokemonId)} onLoad={onImageLoad} blur={blur} alt="name.png" />
    </Header>
);

CardHeader.propTypes = {
    blur: PropTypes.bool.isRequired,
    genus: PropTypes.string,
    name: PropTypes.string,
    onImageLoad: PropTypes.func.isRequired,
    pokemonId: PropTypes.number.isRequired,
};

CardHeader.defaultProps = {
    genus: '',
    name: '',
};
