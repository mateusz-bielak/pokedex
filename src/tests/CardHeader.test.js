import React from 'react';
import { shallow } from 'enzyme';

import { CardHeader } from '../CardHeader';

const props = [
    {
        blur: false,
        genus: 'Seed PokÃ©mon',
        name: 'Bulbasaur',
        onImageLoad: jest.fn(),
        pokemonId: 1,
    },
    {
        blur: false,
        genus: '',
        name: '',
        onImageLoad: jest.fn(),
        pokemonId: 2,
    },
];

it('render correctly CardHeader component', () => {
    const TextInputComponent = shallow(<CardHeader {...props[0]} />);
    expect(TextInputComponent).toMatchSnapshot();
});

it('use default props for genus and name', () => {
    const component = shallow(<CardHeader {...props[1]} />);

    expect(component.find('Styled(span)').text()).toBe('');
    expect(component.find('span').text()).toBe('');
});
