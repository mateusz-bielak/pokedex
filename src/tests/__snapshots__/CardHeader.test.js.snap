// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly CardHeader component 1`] = `
<Styled(header)>
  <Styled(div)>
    <Styled(span)>
      Bulbasaur
    </Styled(span)>
    <span>
      Seed Pok√©mon
    </span>
  </Styled(div)>
  <Styled(img)
    alt="poke"
    blur={false}
    onLoad={[MockFunction]}
    src="https://pokeres.bastionbot.org/images/pokemon/1.png"
  />
</Styled(header)>
`;
